gprofiler/containers_client.py

```diff
@@ -25,20 +25,23 @@
 
 logger = get_logger_adapter(__name__)
 
+_containers_client: Optional[ContainersClient] = None
+
 
 class ContainerNamesClient:
     def __init__(self) -> None:
+        global _containers_client
         try:
-            self._containers_client: Optional[ContainersClient] = ContainersClient()
-            logger.info(f"Discovered container runtimes: {self._containers_client.get_runtimes()}")
+            if _containers_client is None:
+                _containers_client = ContainersClient()
```

Comment:
Let's extract this to a method `get_containers_client` which will be locked (to avoid any concurrency issues) and maintain the cache.

Then, `ContainerNamesClient` runs `self._containers_client = get_containers_client()`.